src/config.rs

```diff
@@ -0,0 +1,1235 @@
+                );
+                Ok(None)
+            }
+            RuleElement::Deny(attrs) => match OptionalOperation::try_from(attrs)? {
+                Some(some) => Ok(Some((Access::Deny, some))),
+                None => Ok(None),
+            },
+        }
+    }
+}
+
+fn rules_try_from_rule_elements(value: Vec<RuleElement>) -> Result<Vec<Rule>> {
+    let mut rules = vec![];
+    for rule in value {
+        let rule = OptionalRule::try_from(rule)?;
+        if let Some(some) = rule {
+            rules.push(some);
+        }
+    }
+    Ok(rules)
+}
+
+pub type Rule = (Access, Operation);
+
+#[derive(Clone, Debug, Default, Deserialize, PartialEq)]
+struct RuleAttributes {
+    #[serde(rename = "@max_fds")]
+    max_fds: Option<u32>,
+    #[serde(rename = "@min_fds")]
+    min_fds: Option<u32>,
+
+    #[serde(rename = "@receive_error")]
+    receive_error: Option<String>,
+    #[serde(rename = "@receive_interface")]
+    receive_interface: Option<String>,
+    /// deprecated and ignored
+    #[serde(rename = "@receive_member")]
+    receive_member: Option<String>,
+    #[serde(rename = "@receive_path")]
+    receive_path: Option<String>,
+    #[serde(rename = "@receive_sender")]
+    receive_sender: Option<String>,
+    #[serde(rename = "@receive_type")]
+    receive_type: Option<MessageType>,
+
+    #[serde(rename = "@send_broadcast")]
+    send_broadcast: Option<bool>,
+    #[serde(rename = "@send_destination")]
+    send_destination: Option<String>,
+    #[serde(rename = "@send_destination_prefix")]
+    send_destination_prefix: Option<String>,
+    #[serde(rename = "@send_error")]
+    send_error: Option<String>,
+    #[serde(rename = "@send_interface")]
+    send_interface: Option<String>,
+    /// deprecated and ignored: https://github.com/dbus2/busd/issues/79
+    #[serde(rename = "@send_member")]
+    send_member: Option<String>,
+    #[serde(rename = "@send_path")]
+    send_path: Option<String>,
+    #[serde(rename = "@send_type")]
+    send_type: Option<MessageType>,
+
+    /// deprecated and ignored
+    #[serde(rename = "@receive_requested_reply")]
+    receive_requested_reply: Option<bool>,
+    /// deprecated and ignored
+    #[serde(rename = "@send_requested_reply")]
+    send_requested_reply: Option<bool>,
+
+    /// deprecated and ignored
+    #[serde(rename = "@eavesdrop")]
+    eavesdrop: Option<bool>,
+
+    #[serde(rename = "@own")]
+    own: Option<String>,
+    #[serde(rename = "@own_prefix")]
+    own_prefix: Option<String>,
+
+    #[serde(rename = "@group")]
+    group: Option<String>,
+    #[serde(rename = "@user")]
+    user: Option<String>,
+}
+
+#[derive(Clone, Debug, Deserialize, PartialEq)]
+#[serde(rename_all = "snake_case")]
+enum RuleElement {
+    Allow(RuleAttributes),
+    Deny(RuleAttributes),
+}
+
+#[derive(Clone, Debug, PartialEq)]
+pub struct SendOperation {
+    pub broadcast: Option<bool>,
+    pub destination: String,
+    pub destination_prefix: String,
```

Comment:
* afaik, you can only specify one of these (at least it doesn't make sense to specify both and we can safely error out on such cases?), so perhaps another enum type is in order?
* should be optional.